services/orders_item_add/lambda_function.py

```diff
@@ -0,0 +1,59 @@
+import os
+import json
+import uuid
+import boto3
+import decimal
+from boto3.dynamodb.conditions import Key, Attr
+from botocore.exceptions import ClientError
+
+# get the service resource
+dynamodb = boto3.resource('dynamodb')
+# get the users table
+ORDERS_TABLE = os.environ['ORDERS_TABLE']
+STOCK_TABLE = os.environ['STOCK_TABLE']
+
+# helper class to convert a DynamoDB item to JSON
+# for details see: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GettingStarted.Python.03.html
+class DecimalEncoder(json.JSONEncoder):
```

Comment:
We should reuse this class if possible

Reply:
agreed

Reply:
if we want to reuse this class, then we will need to pack an additional dependecy for our lambdas most probably which will increase our build times a bit, so that's why I went for using this class within the lambdas

Reply:
We could just have a single file containing the Encoder/Decoder, right? I assume that won’t be that big of a problem w.r.t. build time.

Reply:
it does increase the build times, the script needs to bundle everything into a .zip, but prior to doing that it has to generate an appropriate requirements file for the .zip, thus it actually needs to spin up a docker container, create a virtual .evn, install all dependencies for the script (and all the libraries/dependencies it uses) and then finall generate the requirements.txt file and .zip everything, there are a lot of things actually going on in the background when using external libraries or other .py scripts, so I think it's easier to keep it simple in this case

Reply:
@cldme I think you are mistaken. For this, we can put it in a separate Python file and specify it as an additional asset, since `aws_lambda.Code.asset` allows us to include an entire directory. CDK will then do the zipping for us.

The complicated building step is only needed if we use external dependencies that rely on native code.

Reply:
Seeing that we have also applied this pattern for other services, we should do it in a separate PR.